contract/tests/test_contract.py

```diff
@@ -133,6 +134,36 @@ def _add_template_line(self, overrides=None):
         vals.update(overrides)
         return self.env['contract.template.line'].create(vals)
 
+    def test_add_modifications(self):
+        self.contract.message_subscribe(
+            partner_ids=self.contract.partner_id.ids,
+            subtype_ids=self.env.ref(
+                'contract.mail_message_subtype_contract_modification'
+            ).ids
+        )
+        self.contract.write({
+            'modification_ids': [
+                (
+                    0,
+                    0,
+                    {
+                        'date': '2020-01-01',
+                        'description': 'Modification 1',
+                    }
+                ),
+                (
+                    0,
+                    0,
+                    {
+                        'date': '2020-02-01',
+                        'description': 'Modification 2',
+                    }
+                )
+            ]
+        })
+        self.assertEqual(len(self.contract.modification_ids.ids), 2)
```

Comment:
This assert doesn't add value,  as it's testing the framework, not the feature. You can search for mail.message linked to the contract ID if you want to check for the sent mail.